ui/src/custom-service-worker.js

```diff
@@ -0,0 +1,38 @@
+const URL_PATTERN =
+  /https?:\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()@:%_\+.~#?&//=]*)/gi;
```

Comment:
What are we trying to match here? Can you maybe add a comment with an explanation of the structure we're expecting, just so humans don't have to parse the RE to understand this...

Reply:
Since the `add` endpoint is expecting a URL, and the text from the share menu often contains more than just the URL (additional text), this extracts the URL from the shared text. I know it is quite unreadable, and probably even incomplete, for URLs with other schemas than HTTP. Do you have another idea for this?